### DISCUSSION TITLE
Add Microsoft ToDo Integration ?

### DISCUSSION THREAD

**Discussion by Vasmares** [BimmerGestalt/AAIdrive#424] [Ideas]
Hi,

I discovered this project only some days ago. Great job!
Works brilliant in my F54 2020 MINI ID5.

Have you considered adding support for Microsoft ToDo ?
When driving along, I sometimes remember that I have to do stuff.
At the moment I send an e-mail to myself. A Microsoft ToDo integration would be awsome.

It also has as REST API
https://docs.microsoft.com/de-de/graph/api/resources/todo-overview?view=graph-rest-1.0

**Comment by hufman** [comment]
A very interesting idea! What sort of integration do you imagine? I haven't used this app before, so tell me how you use it and how it would work it the car! Your car might already provide voice memos, but I can understand how the better integration into your existing data would be preferred.
I read that the previous Google Assistant integration has ceased, which is a disappointment.
I feel this idea might be a good candidate for an [Addon app](https://github.com/BimmerGestalt/IDriveConnectAddons).

**Reply by Vasmares** [reply]
Hi,

I have attached a screenshot of the basic GUI of the Windows App.
![todo_gui](https://user-images.githubusercontent.com/10863304/135730425-0b6645f7-2d66-4ab5-864e-0efaf88b27bb.png)

When adding this to the car, I dont think it will be possible to implement all features, but I think the "important basics" should work fine:
 I thought of this:
 One more App in the Connected Menu App list
![mini_connected_todo](https://user-images.githubusercontent.com/10863304/135730403-f09035f3-45bb-4f5f-ba5a-054ed7b81501.png)

When selecting the ToDo App, show some options and your lists incase you want to add it to a secific list like:

\ + New Entry
\ ------------------
\ Important Stuff


I think adding a new entry can ge done with the default MINI keyboard (touch / controller or voice input)
And when clicking on a list that is already there you can see whats in there.

Do you get what I mean ?

**Reply by hufman** [reply]
I like this idea! It would really show off my favorite strength of the car's system, how your personal life gets integrated into the car. I've added it as an idea in the Addons repo for future planning! Thank you so much!

**Reply by Vasmares** [reply]
No Problem :D
Glad to share Ideas

**Reply by bogdan-calapod** [reply]
I concur - really cool idea! :D 

@hufman as a starting point, the ToDo API is basically the Outlook Tasks API. It's pretty basic - I can start working on a Kotlin "wrapper" that can be then called from the addon integration point, if you give me some starting pointers :D

**Reply by Vasmares** [reply]
Would be awsome.
Glad to be a beta tester :D

**Reply by penroseg** [reply]
I'm also happy to Beta test this.  It's something I have always wanted as ideas do come to you when driving and you want to take a note of them.  I also listen to a lot of podcasts and often hear something I want to try later and want to quickly make a note of it.  I've tried using various services (Alexa, Google Keep, Wunderlist etc) to add to a tasks list while driving but then it's a hassle to sync those to Outlook/ToDo tasks so ended up parsing and re-entering.

**Comment by Vasmares** [comment]
Did you find time to try this ?

**Reply by hufman** [reply]
No there has been no progress on this.
